.github/workflows/test-runner-pr-integration.yml

```diff
@@ -0,0 +1,103 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Test Runner PR Integration Tests
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  test-runner-integration:
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:12.3
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+        - 5432:5432
```

Comment:
nit: consider making variables for the ports for readability.

Reply:
I'm inclined to leave this since it's the default postgres port, but I could go either way

Reply:
do we even need  to specify it then?